data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/OperatorConfiguration.java

```diff
@@ -110,12 +118,35 @@ public GenericRegexMatchOperator regexNotEqualOperator() {
 
     @Bean
     public GenericEqualOperator equalOperator() {
-        return new GenericEqualOperator(DataPrepperExpressionParser.EQUAL, equals);
+        final BiFunction<Object, Object, Boolean> floatEquals = (lhs, rhs) -> ((Number)lhs).floatValue() == ((Number) rhs).floatValue();
```

Comment:
I'm not sure I see the value of having all this code here and not in the equals class. As the number of cases expands, we'd need to cram even more into this function.